x-pack/auditbeat/processors/sessionmd/processdb/db.go

```diff
@@ -406,9 +407,15 @@ func (db *DB) InsertExit(exit types.ProcessExitEvent) {
 	defer db.mutex.Unlock()
 
 	pid := exit.PIDs.Tgid
-	delete(db.processes, pid)
```

Comment:
So, these will never be deleted, right? Won't that grow forever?

Reply:
This will grow to `PID_MAX`, and then start to roll-over and replace older entries with new entries when new processes are created with reused PIDs.

Still I agree this could be unnecessarily large, I'll look into adding time-based eviction

Reply:
I am writing this here as a source of inspiration https://github.com/elastic/beats/blob/main/libbeat/processors/add_process_metadata/cache.go#L86. In my perfect world this processor would be able to use the same process cache with the `add_process_metadata` processor, but definitely not for this one

Reply:
I've change to use a goroutine that will run on a timer and clean up exited processes after at least 10s after they have exited. I think this should be more than enough to run the event enrichment, and then they aren't needed again.

@pkoutsovasilis I used a heap/priority queue instead of the list like in that other processor, because I think that times when the list of processes to remove gets very large will also be at times when CPU usage is very high, so it's probably worth it to use the more efficient data structure. For the `add_process_metadata` processor specifically, I don't think the processors will often run at the same time, since they mostly collect the same data, there's not going to be a need to run both

I've also now added a few test cases to exercise pid reuse scenarios.

Reply:
sure @mjwolf this was just a pointer to how an existing code is doing it no objection in the approach you will choose.

> For the add_process_metadata processor specifically, I don't think the processors will often run at the same time, since they mostly collect the same data, there's not going to be a need to run both 

This is actually a really nice point and that's the reason why initially, when I first heard about this processor, I assumed this is gonna end up "extending" the `add_process_metadata` one but evidently my assumption was incorrect